posts/2024-05-01-enabling-rust-lld-on-linux.md

```diff
@@ -0,0 +1,93 @@
+---
+layout: post
+title: Faster linking times on nightly on linux with `rust-lld`
+author: Rémy Rakic
+team: the compiler performance working group <https://www.rust-lang.org/governance/teams/compiler#team-wg-compiler-performance>
+---
+
+TL;DR: rustc will use `rust-lld` by default on `x86_64-unknown-linux-gnu` on nightly to
+significantly reduce linking times.
+
+#### Some context
+
+Linking time is often a big part of compilation time. When rustc needs to build a binary or a shared
+library, it will usually call the default linker installed on the system to do that (this can be
+changed on the command-line or by the target for which the code is compiled).
+
+The linkers do an important job, with concerns about stability, backwards-compatibility and so on.
+For these and other reasons, on the most popular operating systems they usually are older programs,
+designed when computers only had a single core. So, they usually tend to be slow on a modern
+machine. For example, when building ripgrep 13 in debug mode on linux, roughly half of the time is
+actually spent in the linker.
+
+There are different linkers, however, and the usual advice to improve linking times is to use one of
+these newer and faster linkers, like LLVM's [`lld`](https://lld.llvm.org/) or Rui Ueyama's
+[`mold`](https://github.com/rui314/mold).
```

Comment:
Would it be worth calling out https://github.com/davidlattimore/wild ? It's not really ready for use, but it's potential is very exciting (and of course it's written in Rust)

Reply:
I think wild is not mature enough to be included in a "use one of these" list.

Reply:
I also think it's a bit too early to mention it. We're all looking forward to when it's ready though :)